src/components/UpsertMovieModal.js

```diff
@@ -20,63 +23,67 @@ export default function UpsertMovieModal({header, movie, handleCloseModal}){
             handleUpdate(movie, data);
         }
 
-        console.log(contextValue);
-
-        contextValue.setMovieRepo(movieRepo.clone());
         handleCloseModal();
     };
     const onError = (errors, e) => console.log(errors, e);
 
     const handleUpdate = (movie, formData)=>{
-        if(formData.title && formData.title !==""){
-            movie.name = formData.title;
+        if(formData.title && formData.title !== ""){
+            movie.title = formData.title;
         }
-        if(formData.genre && formData.genre !==""){
-            movie.genres = [{name: formData.genre}];
+        
+        if(
+                formData.genre && 
+                formData.genre !== "" && 
+                formData.genre !== "All"
+            ){
+            movie.genres = [formData.genre];
         }
 
         if(formData.url && formData.url !==""){
-            movie.url = formData.url;
+            movie.poster_path = formData.url;
         }
 
         if(formData.rating && formData.rating !==""){
-            movie.rating = formData.rating;
+            movie.vote_average = Number(formData.rating);
         }
         if(formData.runtime && formData.runtime !==""){
-            movie.runtime = formData.runtime;
+            movie.runtime = Number(formData.runtime);
         }
 
         if(formData.overview && formData.overview !==""){
             movie.overview = formData.overview;
         }
 
-        movieRepo.updateMovie(movie);
+        updateMovie(movie)
     };
 
 
     const handleCreate = (formData)=>{
         const newMovie = {};
-        newMovie.name = formData.title;
-        newMovie.genres = [{name: formData.genre}];
-        newMovie.url = formData.url;
-        newMovie.rating = formData.rating;
-        newMovie.runtime = formData.runtime;
+        newMovie.title = formData.title;
+        newMovie.genres = [formData.genre];
+        newMovie.poster_path = formData.url;
+        newMovie.vote_average = Number(formData.rating);
+        newMovie.runtime = Number(formData.runtime);
         newMovie.overview = formData.overview;
-        movieRepo.addMovie(newMovie);
+        createMovie(newMovie)
     };
 
     const defaultFormData = {
         title: "best movie",
-        url: "https://",
-        rating: "7.8",
-        runtime: "minutes"
+        url: "https://default-path",
+        rating: 7.8,
+        runtime: 100,
+        overview: "Lorem ipsum"
```

Comment:
Here you use default values for the form, but actually it isn't correct. Each form tag has `placeholder` attribute, and when you use it the value inside doesn't used when you submit the form. Right now you can save record with default values

Reply:
done